libpod/runtime_pod_infra_linux.go

```diff
@@ -146,11 +146,32 @@ func (r *Runtime) makeInfraContainer(ctx context.Context, p *Pod, imgName, rawIm
 			options = append(options, WithExitCommand(p.config.InfraContainer.ExitCommand))
 		}
 	}
+
+	for _, ctl := range r.config.Containers.DefaultSysctls {
+		sysctl := strings.SplitN(ctl, "=", 2)
+		if len(sysctl) < 2 {
+			return nil, errors.Errorf("invalid default sysctl %s", ctl)
+		}
+		g.AddLinuxSysctl(sysctl[0], sysctl[1])
+	}
+
 	g.SetRootReadonly(true)
 	g.SetProcessArgs(infraCtrCommand)
 
 	logrus.Debugf("Using %q as infra container command", infraCtrCommand)
 
+	mapopt, err := util.ParseIDMapping(namespaces.UsernsMode(p.config.InfraContainer.Userns.String()), []string{}, []string{}, "", "")
+	if err != nil {
+		return nil, err
+	}
+	user, err := specgen.SetupUserNS(mapopt, p.config.InfraContainer.Userns, &g)
+	if err != nil {
+		return nil, err
+	}
+	if user != "" {
+		options = append(options, WithUser(user))
```

Comment:
Do we need this? I think this is only for keep-id and it doesn't really make sense for infra containers - other containers in the pod won't inherit it.

Reply:
The pod fails to create without this when using keep-id. Should I make it a condition to only do this when userns is set to keep-id?
This is the error I get 
```
Error: error adding Infra Container: please set User explicitly via WithUser() instead of in OCI spec directly: invalid argument
```

Reply:
You should be able to clear the User field in the OCI spec (set it to "" or root:root) and that will clear the error

Reply:
@mheon we want containers inside a pod to inherit the user from the infra container when --userns=keep-id and the container was not created with the --user flag. This kind of links to the discussion we had in https://github.com/containers/podman/pull/10589#discussion_r677718549.
I think for that case, we should leave this as is instead of always adding `root:root` to the infra container, wdyt?

Reply:
If we're using it to have other containers inherit it, that's probably OK, but I'm still worried that running infra images as non-root users may not be very durable with user-supplied infra images.

Reply:
Could this be a runc versus crun issue?

Reply:
@mheon I will open up a follow PR for the user work. Setting the user to root:root always for the infra container is causing some other issues. Can we get this in for now while I debug that?